apps/remix/app/components/general/document/document-search.tsx

```diff
@@ -30,8 +30,12 @@ export const DocumentSearch = ({ initialValue = '' }: { initialValue?: string })
   );
 
   useEffect(() => {
-    handleSearch(searchTerm);
-  }, [debouncedSearchTerm]);
+    const currentQueryParam = searchParams.get('query') || '';
+
+    if (debouncedSearchTerm !== currentQueryParam) {
```

Comment:
ensures that the search params are only updated when there's an actual change to prevent unnecessary URL updates. it prevents the double clicking when navigating back to the previous page